### DISCUSSION TITLE
Nextjs: getInitialProps axios header config typescript

### DISCUSSION THREAD

**Discussion by TarikNasraoui** [vercel/next.js#37422] [Help] [answered]
Hi,
How i can satisfy typescript here: I have a problem here << headers: req?.headers >>, header type does not satisfy ts

please see screenshot below:

<img width="794" alt="Capture d’écran 2022-06-02 à 23 42 45" src="https://user-images.githubusercontent.com/30218125/171819211-b6fc3f15-b27f-4060-9160-4b0b6f473248.png">

<img width="918" alt="Capture d’écran 2022-06-02 à 23 46 03" src="https://user-images.githubusercontent.com/30218125/171818725-56fa1d39-03c2-4e89-a7c3-5061fc98c927.png">

normally I have to change NextPageContext or add another type of axios?

**Comment by icyJoseph** [comment, marked as answer]
~~What axios version are you using?~~ This has been fixed about 20 days ago in their library typings. 

Yeah this, https://github.com/axios/axios/pull/4334 PR fixed it but it hasn't been released.

### A hack

Please, once axios releases their update, you should remove this:

```ts
const axiosHeaders = Array.from(Object.entries(req?.headers ?? {})).reduce<Record<string, string | string[]>>(
    (prev, [key, value]) => {
        return {...prev, [key]: typeof value === 'undefined' ? `${value}` : value};
    },
    {}
);


```

This is basically removing the `undefined` value in favour of an empty string, which will make TS happy, but your code and team workers, sad 😅. (There's most likely other ways to do this, I am just working a lot with reducers this morning)

**Reply by TarikNasraoui** [reply]
I use this version:   "axios": "^0.27.2",
i think this this is the last version

**Reply by icyJoseph** [reply]
Checkout my updated answer. I had missed that they hadn't released since April 27th.

**Reply by TarikNasraoui** [reply]
thank you @icyJoseph  :)
